.ci/generate_bear_requirements.py

```diff
@@ -167,6 +176,46 @@ def get_pip_requirements(requirements):
 
     return pip_requirements
 
+def deepupdate(target, src):
+    for key, value in src.items():
+        if isinstance(value, list):
+            if not key in target:
+                target[key] = copy.deepcopy(value)
```

Comment:
It is simpler if you always do this, irrespective of the type
```
if not key in target:
    target[key] = copy.deepcopy(value)
```

Then in the else you can do the different update mechanisms.
That makes it more clear that this is like a deepcopy for any new keys, but behaves differently for other keys, and raise an exception if there is an update required for an unknown type as we will need to add support for it whenever that happens.